### PR TITLE
Add dropEffect getter and setter on DataTransfer

### PR DESCRIPTION
**Pull request by fehrenbach**

### CODE REVIEW

File: src/DOM/HTML/Event/DataTransfer.purs [hunk 1]
```diff
@@ -61,3 +66,27 @@ setData
...
+
+--| Gets the data transfer object's drop effect.
+dropEffect :: forall eff. DataTransfer -> Eff (dom :: DOM | eff) DropEffect
+dropEffect dt = do
+  de <- dropEffectImpl dt
+  pure $ unsafePartialBecause "No other values are possible https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/dropEffect" case de of
+    "copy" -> Copy
+    "move" -> Move
+    "link" -> Link
+    "none" -> None
```
Comment (garyb) [inline src/DOM/HTML/Event/DataTransfer.purs]: `unsafePartialBecause` is problematic (https://github.com/purescript/purescript-partial/issues/7), could you use `crashWith` in an `_` case, and then `unsafePartial` the `case` instead please?

File: src/DOM/HTML/Event/DataTransfer.purs [hunk 2]
```diff
@@ -61,3 +66,27 @@ setData
   -> DataTransfer
   -> Eff (dom :: DOM | eff) Unit
 setData (MediaType format) dat dt = setDataImpl format dat dt
+
+foreign import dropEffectImpl :: forall eff. DataTransfer -> Eff (dom :: DOM | eff) String
+
+data DropEffect = None | Copy | Move | Link
```
Comment (garyb) [inline src/DOM/HTML/Event/DataTransfer.purs]: Could you add derived instances for `Eq` and `Ord` for this too please?

**Review by garyb — CHANGES_REQUESTED** [review]
Your PR to Halogen reminded me of this one that has been outstanding for ages. Sorry about the delay!

**Comment by fehrenbach** [comment]
No worries. What kind of heathen makes pull requests on Christmas Day anyways? ;)

Thanks for the review. I addressed your remarks and sorted the cases while I was at it.

**Comment by garyb** [comment]
Thanks!

**Review by garyb — APPROVED** [review]

**Pull request merged**